File: backend/models/advertisement.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field

class Advertisement(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    title: str
    subtitle: str
    button_text: str = "Explore Now"
    image_url: str
    redirect_link: Optional[str] = None
    ad_type: str = "Hero Banner" # Hero Banner, Offer, Announcement
    display_order: int = 0
    status: str = "active" # active, inactive
    start_date: datetime
    end_date: datetime
    click_count: int = 0
    impression_count: int = 0
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        populate_by_name = True

File: frontend/lib/features/home/models/advertisement.dart
class Advertisement {
  final String id;
  final String title;
  final String subtitle;
  final String buttonText;
  final String imageUrl;
  final String? redirectLink;
  final String adType;
  final int displayOrder;
  final String status;

  Advertisement({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.buttonText,
    required this.imageUrl,
    this.redirectLink,
    required this.adType,
    required this.displayOrder,
    required this.status,
  });

  factory Advertisement.fromJson(Map<String, dynamic> json) {
    return Advertisement(
      id: json['_id'] ?? json['id'] ?? '',
      title: json['title'] ?? '',
      subtitle: json['subtitle'] ?? '',
      buttonText: json['button_text'] ?? 'Explore Now',
      imageUrl: json['image_url'] ?? '',
      redirectLink: json['redirect_link'],
      adType: json['ad_type'] ?? 'Hero Banner',
      displayOrder: json['display_order'] ?? 0,
      status: json['status'] ?? 'active',
    );
  }
}
